plugins/datetime/msd-datetime-mechanism.c

```diff
@@ -450,199 +497,360 @@ msd_datetime_mechanism_set_timezone (MsdDatetimeMechanism  *mechanism,
-        return TRUE;
+static void
+msd_datetime_mechanism_init (MsdDatetimeMechanism *mechanism)
+{
+        mechanism->priv = msd_datetime_mechanism_get_instance_private (mechanism);
+        mechanism->priv->skeleton = mate_settings_date_time_mechanism_skeleton_new ();
 }
 
-gboolean
-msd_datetime_mechanism_set_hardware_clock_using_utc (MsdDatetimeMechanism  *mechanism,
-                                                     gboolean               using_utc,
-                                                     DBusGMethodInvocation *context)
+static void
+msd_datetime_mechanism_dispose (GObject *object)
 {
-        GError *error;
+        MsdDatetimeMechanism *mechanism;
+
+        g_return_if_fail (object != NULL);
+        g_return_if_fail (MSD_DATETIME_IS_MECHANISM (object));
 
-        error = NULL;
+        mechanism = MSD_DATETIME_MECHANISM (object);
 
-        if (!_check_polkit_for_action (mechanism, context,
-                                       "org.mate.settingsdaemon.datetimemechanism.configurehwclock"))
-                return FALSE;
+        g_return_if_fail (mechanism->priv != NULL);
 
-        if (g_file_test ("/sbin/hwclock",
-                         G_FILE_TEST_EXISTS | G_FILE_TEST_IS_REGULAR | G_FILE_TEST_IS_EXECUTABLE)) {
-                int exit_status;
-                char *cmd;
-                cmd = g_strdup_printf ("/sbin/hwclock %s --systohc", using_utc ? "--utc" : "--localtime");
-                if (!g_spawn_command_line_sync (cmd, NULL, NULL, &exit_status, &error)) {
-                        GError *error2;
-                        error2 = g_error_new (MSD_DATETIME_MECHANISM_ERROR,
-                                              MSD_DATETIME_MECHANISM_ERROR_GENERAL,
-                                              "Error spawning /sbin/hwclock: %s", error->message);
-                        g_error_free (error);
-                        dbus_g_method_return_error (context, error2);
-                        g_error_free (error2);
-                        g_free (cmd);
-                        return FALSE;
-                }
-                g_free (cmd);
-                if (WEXITSTATUS (exit_status) != 0) {
-                        error = g_error_new (MSD_DATETIME_MECHANISM_ERROR,
-                                             MSD_DATETIME_MECHANISM_ERROR_GENERAL,
-                                             "/sbin/hwclock returned %d", exit_status);
-                        dbus_g_method_return_error (context, error);
+        if (mechanism->priv->skeleton != NULL)
+        {
+            GDBusInterfaceSkeleton *skeleton;
+
+            skeleton = G_DBUS_INTERFACE_SKELETON (mechanism->priv->skeleton);
+            g_dbus_interface_skeleton_unexport (skeleton);
+            g_clear_object (&mechanism->priv->skeleton);
+        }
+
+        if (mechanism->priv->bus_name_id > 0)
+        {
+            g_bus_unown_name (mechanism->priv->bus_name_id);
+            mechanism->priv->bus_name_id = 0;
```

Comment:
Use NxW spaces wide to indent, where N is the indent level and W is the number of spaces of the 1st indent level.